mungegithub/mungers/submit-queue.go

```diff
@@ -287,13 +287,13 @@ func calcMergeRate(oldRate float64, last, now time.Time) float64 {
 
 // updates a smoothed rate at which PRs are merging per day.
 // returns 'Now()' and the rate.
-// Should be called once after every merge. Also updates sq.totalMerges.
+// Note that this does not update sq.lastMergeTime-- that should be done with
+// every merge, but this function should not be called on retest-not-required
+// merges.
 func (sq *SubmitQueue) updateMergeRate() {
 	now := sq.clock.Now()
 
 	sq.mergeRate = calcMergeRate(sq.mergeRate, sq.lastMergeTime, now)
-	sq.lastMergeTime = now
```

Comment:
> throw off the rate calculation when the next NON-retest-not-required comes through

I think that's a feature, not a bug. 

I think your math makes numbers that are generally in the right ballpark and we should not throw it away yet :)